sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/IncrementalExecution.scala

```diff
@@ -57,7 +59,9 @@ class IncrementalExecution(
     val prevOffsetSeqMetadata: Option[OffsetSeqMetadata],
     val offsetSeqMetadata: OffsetSeqMetadata,
     val watermarkPropagator: WatermarkPropagator,
-    val isFirstBatch: Boolean)
+    val isFirstBatch: Boolean,
+    val currentCheckpointUniqueId:
+      MutableMap[Long, Array[String]] = MutableMap[Long, Array[String]]())
```

Comment:
Can we add comments on what are these unique Ids map to? I believe key is partition?

Reply:
If key is operator Id how do we store partitions?

Reply:
also better name it `currentStateUniqueId` as it is only related to state store not general checkpoint

Reply:
I'm also confused by this. When I sketched an implementation of your proposal in my head, my assumption would be that `IncrementalExecution` would get just an ID, perhaps a _single_ Long, that would correspond to the ID that it would bake into the physical plan sent to executors. So why is a map needed?

Reply:
I'll add a comment, but it is basically operatorID->partitionID->checkpointID